lib/bemxjst/index.js

```diff
@@ -117,10 +117,6 @@ BEMXJST.prototype.compile = function compile(code) {
 
 BEMXJST.prototype.recompileInput = function recompileInput(code) {
   var args = BEMXJST.prototype.locals;
-  // Reuse function if it already has right arguments
-  if (typeof code === 'function' && code.length === args.length)
-    return code;
-
```

Comment:
Because this case is fantastic:

```
bemxjst.compile(function(block, elem, mod, addMods, content, …) { … })
```

Nobody wants to enumerate all modes.